### DISCUSSION TITLE
Is pywinauto still being supported (updated)?

### DISCUSSION THREAD

**Discussion by ikommkophee** [pywinauto/pywinauto#1357] [Q&A]
Hi all. The discussions section shows the latest post was in 2022 so wanted to get an idea if pywinauto  is still being supported and/or updated with new features/bug fixes? This tool has been a game-changer for me in aiding me in my regression testing (though I'm still learning) and would love to see it continue being developed. And would love to eventually ask the authors about how to automate some things that have me stumped. Cheers!

**Comment by vasily-v-ryabov** [comment]
Hi @ikommkophee currently pywinauto is rarely maintained due to my load on main job and with other personal things. There are few big pull requests with macOS support and DLL injection approach for .NET apps. I want to finish these things one day. Hopefully I will have some time during New Year holidays to read all notifications and do something. Please tell your ideas and ask questions. I can't promise quick replies till ~December, 25, but maybe someone else can reply eventually.

**Reply by ikommkophee** [reply]
Thank you very much for the response, Vasily. Much appreciated.
I'm hoping you (or anyone else) can help me with the issue I'm currently stuck on. 

I'm trying to access a separate column on a dialog window to select a specific item but so far I can only use type_keys() to select the item on the first column. I'd like to be able to select an item in a separate column (date column for example) to have finer control. 

![screenshot](https://github.com/pywinauto/pywinauto/assets/48231085/2e24d4e3-196d-450f-8301-b108432cc261)

Using the methods for ListBox and ListView doesn't work and gives me an error. 
![Capture02](https://github.com/pywinauto/pywinauto/assets/48231085/cc85dd33-a70f-4bf9-8120-82ddda6f9396)

I'm starting to wonder if there is actually a method that can select an item in the listpicker. This is frustrating since a big part of the app we use is centered on using list windows like this one.
Might you have some insight on this, or can you point me in the right direction? Any input would help out a lot. In the meantime I'll keep at this and see if I'll get lucky. 

Thank you very much for your time and Happy Holidays.

Additional screenshots for reference: 
**listpicker control:**
![03](https://github.com/pywinauto/pywinauto/assets/48231085/7b214f1e-5773-4755-8800-3303a47b5695)

**print_control_identifiers output of the dialog window:**
![Capture](https://github.com/pywinauto/pywinauto/assets/48231085/c8b343a6-1f3c-4b9f-92fa-1d0f93843736)

**Reply by junkmd** [reply]
I would like to ask you first, which one are you using for the `backend`, `"uia"` or `"win32"`?

**Reply by ikommkophee** [reply]
Hey! This'll be win32. On the topic of backend = , while I've determined that it is indeed win32, what kind of effect it would have if I took it out and just left it as Application()? So far I haven't noticed anything breaking the script. 
Thanks for your time!

**Reply by junkmd** [reply]
You are right, in `Application()`, the backend to use is `"win32"` by default.

There are other backends in `pywinauto` besides `"win32"` that can manipulate the GUI of the WIndows environment.
Have you tried to see if you can manipulate that GUI with `Application(backend="uia")`?

**Reply by ikommkophee** [reply]
hey webmd,
Yes, at some point I've tried 'uia' but encountered issues even with connecting to the main app dialog. That's why I've defaulted to 'win32' ever since. For this specific error, I used 'click' to 'force' the focus on the main app ribbon. Otherwise, using the 'click_input' function give me an error like the one below:

using click() on 'uia': 
![image](https://github.com/pywinauto/pywinauto/assets/48231085/aaaed306-2f9a-4e71-91a3-ea612aef2923)

click_input() on 'win32':
![image](https://github.com/pywinauto/pywinauto/assets/48231085/c4172a4a-6e6b-48b9-8e8b-50a745152772)

Hope this helps.
Cheers!

**Reply by junkmd** [reply]
Hi,

I understand why you are using `backend='win32'`.

Have you tried such code as the following?
```py
focus_window = main_app.AfxMDIFrame140u
focus_window.click_input()
focus_window.type_keys('^b')
```

I want to see more detailed code from the retrieval of the main window until the error occurs.

If there are no NDA problems, I might be able to give more useful advice if you could take a video screenshot of your script from startup to the error and share it with the community.

**Reply by ikommkophee** [reply]
Yes, that was actually my original script for opening the necessary dialog window. Only reason why I did it like it is right now so that I could shorten the code ;). 

![image](https://github.com/pywinauto/pywinauto/assets/48231085/1da5c186-0fb2-4361-aeea-6096f1dd577a)

But my main concern is trying to, and where I'm encountering difficulty, is to select an item from a window with listview element:
![image](https://github.com/pywinauto/pywinauto/assets/48231085/15489151-a8c5-469e-b888-d3273f4ad969)

Right now I can only access this list by the first column and nothing else. My code is:
**var.listpicker.type_keys('18 All Dielectric')**

I tried using **var.listpicker.HeaderDescription.type_keys('some text here ')**

but only ended up with the list window open but no item selected. The code however shows it as being completed: 
![image](https://github.com/pywinauto/pywinauto/assets/48231085/291299a6-2658-4db3-8242-afa11248830b)

That tells me that the type_keys function went through, just not sure where though. 
It's the same result when I do this: 
![image](https://github.com/pywinauto/pywinauto/assets/48231085/761b13fa-53ed-42d1-a60d-bd38a0ed410c)

As for your request for a video screenshot, I'll try to get a video recording as this is a company computer and our access to certain apps that might be used for recording is monitored. But I hope the info I've provided is enough that the pieces can be put together. 

Thanks!

**Reply by junkmd** [reply]
What do you see when you run `print(scdef.listpicker.items())`?

Are they list of `_listview_item`?

**Reply by ikommkophee** [reply]
This is what it gives me:

![image](https://github.com/pywinauto/pywinauto/assets/48231085/0ff397f4-36f6-4e9e-834d-7593387fe938)

I think I've also tried that function (or some variation of it using .items with no success :( )
I wonder if this'll help some, this is taken from the Accessibility Insight for Windows tool:

the list has a title of '', basically none
![image](https://github.com/pywinauto/pywinauto/assets/48231085/0b651a70-b59e-4d1f-9685-d577d0ba67ef)

and the list items and text are there, I just have no way of accessing those:
![image](https://github.com/pywinauto/pywinauto/assets/48231085/a593090e-d0cb-47ec-90fd-f83d28b7e7b6)
